lib/models/MatrixAction.js

```diff
@@ -26,6 +30,47 @@ function MatrixAction(type, text, htmlText, timestamp) {
     this.htmlText = htmlText;
     this.ts = timestamp || 0;
 }
+
+MatrixAction.prototype.formatMentions = function(nickUserIdMap) {
+    const regexString = "(" +
+        Object.keys(nickUserIdMap).map((value) => escapeStringRegexp(value)).join("|")
+        + ")";
+    const userRegex = new RegExp(regexString, "igm");
+    const matched = new Set(); // lowercased nicknames we have matched already.
+    let match;
+    let i = 0;
+    while ((match = userRegex.exec(this.text)) !== null && i < MAX_MATCHES) {
+        i++;
+        let matchName = match[1];
+        // Deliberately have a minimum length to match on, so we don't match smaller nicks accidentally.
+        if (matchName.length < MIN_LENGTH_TO_MATCH || matched.has(matchName.toLowerCase())) {
+            continue;
+        }
+        let userId = nickUserIdMap[matchName];
+        if (userId === undefined) {
+            // We might need to search case-insensitive.
+            const nicks = Object.keys(nickUserIdMap).filter((nick) =>
```

Comment:
can't we just do `userId = nickUserIdMap[matchName.toLowerCase()];` ?

Reply:
Nicks are case sensitive, so we'd need to maintain two maps.

Reply:
Oh I see now, I'm blind. Should we use `find` instead to avoid going over the entire list in most cases?